plugins/janus_sip.c

```diff
@@ -2493,9 +2499,16 @@ static void *janus_sip_handler(void *data) {
 			}
 			g_atomic_int_set(&session->hangingup, 0);
 			session->status = janus_sip_call_status_inviting;
-			/* Create a random call-id */
-			char callid[24];
-			janus_sip_random_string(24, (char *)&callid);
+			char *callid;
+			json_t *request_callid = json_object_get(root, "call_id");
+			/* Take call-id from request, if it exists */
+			if (request_callid) {
+				callid = g_strdup(json_string_value(request_callid));
+			/* If call-id does not exist in request, create a random one */
```

Comment:
The comment is in the wrong place, please put it inside the `else` and properly indented (same in other plugin, I guess).